library/core/src/marker.rs

```diff
@@ -247,6 +247,7 @@ marker_impls! {
 ///
 /// const CFN: Wrap<fn(&())> = Wrap(higher_order);
 ///
+/// #[allow(pointer_structural_match)]
```

Comment:
FWIW these entire trait docs are outdated; that's tracked in https://github.com/rust-lang/rust/issues/115881.